HealthcareSystemFrontends/GraphicalEditor/MainWindow.xaml.cs

```diff
@@ -831,7 +837,73 @@ private void EmergencyAppointmentSearchResultsDataGrid_SelectionChanged(object s
 
         private void ScheduleEmergencyAppointmentButton_Click(object sender, RoutedEventArgs e)
```

Comment:
This function got really big, so it would be a good idea to refactor it.

Reply:
Yes, we did, I will try to do that soon... Maybe not in this PR, but definitely will do it.